migration-engine/connectors/sql-migration-connector/src/sql_database_migration_inferrer.rs

```diff
@@ -68,16 +69,18 @@ fn infer_database_migration_steps_and_fix(
     to: &SqlSchema,
     schema_name: &str,
     sql_family: SqlFamily,
-) -> SqlResult<Vec<SqlMigrationStep>> {
+) -> SqlResult<(Vec<SqlMigrationStep>, Vec<SqlMigrationStep>)> {
     let diff: DatabaseSchemaDiff = DatabaseSchemaDiffer::diff(&from, &to);
     let is_sqlite = sql_family == SqlFamily::Sqlite;
 
-    if is_sqlite {
-        fix_stupid_sqlite(diff, &from, &to, &schema_name)
+    let corrected_steps = if is_sqlite {
+        fix_stupid_sqlite(diff, &from, &to, &schema_name)?
     } else {
         let steps = delay_foreign_key_creation(diff);
-        fix_id_column_type_change(&from, &to, schema_name, steps)
-    }
+        fix_id_column_type_change(&from, &to, schema_name, steps)?
+    };
+
+    Ok((DatabaseSchemaDiffer::diff(&from, &to).into_steps(), corrected_steps))
```

Comment:
I'm not sure about diffing twice, but I preferred avoiding diving into sort-of unrelated known I haven't seen in this PR.

Reply:
That's not an issue 👍